static/js/components/CompactPostDisplay_test.js

```diff
@@ -71,6 +71,19 @@ describe("CompactPostDisplay", () => {
     assert.isNotEmpty(authoredBy.substring(post.author_name.length))
   })
 
+  it("should not display headline span if author headline is null", () => {
+    post.author_headline = null
+    const wrapper = renderPostDisplay({ post })
+    assert.isNotOk(wrapper.find(".author-headline").exists())
+  })
+
+  it("should display headline span with correct text", () => {
+    post.author_headline = "My headline"
+    const wrapper = renderPostDisplay({ post })
+    const headlineSpan = wrapper.find(".author-headline")
+    assert(headlineSpan.text().includes("My headline"))
+  })
```

Comment:
2 cases above can be parametrized into 1:

```javascript
;[
  ["My headline", true],
  [null, false],
].forEach(([headlineText, expElementExists]) => {
  it(`${shouldIf(expElementExists) display headline span when text=${String(headlineText)}`, () => {
    post.author_headline = headlineText
    const wrapper = renderPostDisplay({ post })
    const headlineSpan = wrapper.find(".author-headline")
    assert.equal(headlineSpan.exists(), expElementExists)
    if (expElementExists) {
      assert(headlineSpan.text().includes(headlineText))
    }
  })
})
```

Reply:
In some previous PR awhile back, I think someone (can't remember who) said to try to avoid conditionally running asserts in tests, which is why I split them up into separate tests here.  Either way works for me, I don't have strong feelings about it as long as the code gets tested.  Let me know if you'd still like me to roll these two tests into one.

Reply:
Generally I think it's good practice to avoid conditionals/"forks" in tests, but here the readability hit is very small, the goals of the test are very clear, and the result is clearer in intent than the same thing spread across two test cases